System/Posix/Fcntl.hsc

```diff
@@ -101,3 +111,68 @@ foreign import capi safe "fcntl.h posix_fallocate"
 fileAllocate _ _ _ = ioError (ioeSetLocation unsupportedOperation
                               "fileAllocate")
 #endif
+
+-- -----------------------------------------------------------------------------
+-- File caching
+
+-- | Performs the @fcntl(2)@ operation on a file-desciptor to get the cache mode.
+--
+-- If the cache mode is 'False', then cache effects for file system reads and
+-- writes are minimised or otherwise eliminated. If the cache mode is 'True',
+-- then cache effects occur like normal.
+--
+-- On Linux, FreeBSD, and NetBSD this checks whether the @O_DIRECT@ file flag is
+-- set.
+--
+-- Throws 'IOError' (\"unsupported operation\") if platform does not support
+-- reading the cache mode.
+--
+-- (use @#if HAVE_O_DIRECT@ CPP guard to detect availability).
```

Comment:
Is this macro available to a user though?

Reply:
Should be through `HsUnix.h`

Reply:
I'd prefer the documentation to say so explicitly, otherwise people will be left wondering who defines them.

Reply:
I've now added a hint to use `#include "HsUnix.h"` to bring the macros into scope